### PR TITLE
Mp/hunter tests

### PR DESCRIPTION
**Pull request by Buvko**

### CODE REVIEW

File: service/src/main/java/cz/muni/fi/pa165/mushrooms/service/facade/VisitFacadeImpl.java [hunk 1]
```diff
@@ -97,9 +93,8 @@ public void updateVisit(VisitDTO visit) {
             //TODO: react to it somehow
         }
 
-
         entityVisit.setHunter(hunter);
-        //entityVisit.setDate(visit.getDate());
+        entityVisit.setDate(LocalDate.parse(visit.getDate()));
```
Comment (bkompis) [inline service/src/main/java/cz/muni/fi/pa165/mushrooms/service/facade/VisitFacadeImpl.java]: what about exception at date parsing? Bad string format is very real :D

**Review by bkompis — CHANGES_REQUESTED** [review]
* exception handling for date format parsing

File: service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+
+    private Map<Long, MushroomHunter> persistedMushroomHunter = new HashMap<>();
+    // stores 'database' size
+    private long dbCounter = 1;
+    int persistedMushroomHunterPreTest;
+
+
+    @Before
+    public void setUp() {
+        // note: for tests here, password hash and password are the same
```
Comment (bkompis) [inline service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java]: are they?

File: service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+        nonPersistedHunter = createHunter("Alphonse", "Elric", "theGoodGuy" , "smt5udp", false);
```
Comment (bkompis) [inline service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java]: what are those things before admin privileges? Random strings?

File: service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+        newHunter = createHunter("Jan", "Jakub", "J2J", "95gh3kew", false);
+        int oldSize = persistedMushroomHunter.size();
+
+        service.registerHunter(newHunter, newHunter.getPasswordHash());
+        assertThat(persistedMushroomHunter.size()).isEqualTo(oldSize + 1);
+    }
+
+    @Test
+    public void createNullMushroomHunter() throws Exception {
+        assertThatThrownBy(() -> service.registerHunter(null, "abcd")).isInstanceOf(EntityOperationServiceException.class);
```
Comment (bkompis) [inline service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java]: DataAccessException (everywhere)

**Review by bkompis — CHANGES_REQUESTED** [review]
* naming: 
   * deleteMushroomHunter, createMushroomHunter... redundant, ofc these are tests for mushroom hunter
    * inconsistent with the rest of the project

* expected exception should be DataAccessException
* date parsing

File: service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java (hunk 3 above)
Comment (Buvko) [reply service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java]: good passwords

**Review by bkompis — APPROVED** [review]

**Pull request merged**